content/blog/2021-07-13-hyperparam-tuning.md

```diff
@@ -0,0 +1,291 @@
+training, and keep up with hyperparameter value updates.
+
+Being able to track all of these changes is important so that you can reproduce
+your experiments without wondering which changes gave you the best model. You
+can go back to any point in your experimenting process to see which changes gave
+you the best results.
+
+In this post, we're going to go through an example of hyperparameter tuning with
+reproducibility using DVC. You can add this to any existing project you're
+working on or start from a fresh project.
+
+## Background on Hyperparameters
+
+Before jumping straight into training and experiments, let's briefly go over
+some background on hyperparameters.
+[Hyperparameters](/doc/command-reference/params) are the values that define your
+model. This includes things like the number of layers in a neural network or the
+learning rate for gradient descent.
+
+These parameters are different from model parameters because we can't get them
+from training our model. They are used to _create_ the model we train with.
+Optimizing these values means running training steps for different kinds of
+models to see how accurate the results are. We can get the best model from
+iterating through different hyperparameter values and seeing how they effect our
+accuracy.
+
+That's why we do hyperparameter tuning. There are a couple common methods that
+we'll do some code examples with: grid search and random search.
+
+## Tuning with DVC
+
+Let's start by talking about DVC a bit because we'll be using it to add
+reproducibility to our tuning process. This is the tool we'll be using to track
+changes in our data, code, and hyperparameters. With DVC, we can add some
+automation to the tuning process and be able to find and restore any really good
+models that emerge.
+
+A few things DVC makes easier to do:
+
+- Letting you make changes without worrying about finding them later
+- Onboarding other engineers to a project
+- Sharing experiments with other engineers on different machines
+
+For hyperparameter tuning, this means you can play with their values without
+losing track of which changes made the best model and also have other engineers
+take a look. We'll do an example of this with grid search in DVC first.
+
+## Working with a DVC project
+
+We're going to be working with an existing NLP project. You can
+[get the code we're working with in this repo](https://github.com/iterative/example-get-started).
+It already has DVC set up, but you can check out
+[the Get Started docs](https://dvc.org/doc/start) if you want to know how the
+DVC pipeline was created.
+
+First make sure you're in a virtual environment with a command similar to this.
+
+```dvc
+$ python -m venv .venv
+```
+
+After you've cloned the repo, install all of the dependencies with this command.
+
+`pip install -r requirements.txt`
+
+You should be able to open your terminal and run an experiment with the
+following command.
+
+`dvc exp run`
```

Comment:
let's use:

```
```dvc
$ dvc exp run
``````

code block

(and in other places, start the line with $ for the syntax highlighter to pick up it